plugins/org.eclipse.n4js.tester/src/org/eclipse/n4js/tester/TestDiscoveryHelper.java

```diff
@@ -168,19 +167,28 @@ public boolean isTestable(final URI location) {
 	 *         tree may be empty.
 	 */
 	public TestTree collectTests(final List<URI> uris) {
-		final ResourceSet resSet = n4jsCore.createResourceSet(Optional.absent());
-		final IResourceDescriptions index = n4jsCore.getXtextIndex(resSet);
```

Comment:
This line used to return an index with TModule information (not an empty index), at least in the UI case.
Has this behavior changed? (in this case, please adjust the API doc of `IN4JSCore#getXtextIndex()`) Or was there some other reason why - before this PR - everything was rebuilt from scratch?

Reply:
`n4jsCore.getXtextIndex(resSet)` tries to find an adapter in the resource set that points to "the index". A newly created resource set doesn't have index information attached. I'm not sure how it worked in the old headless compiler, but it's perfectly reasonable to find an empty index in a an empty resource set in the standalone model. That's why we need to pass it into this method.

I'm _still_ not sure if it's a good idea to have `n4jsCore.createResourceSet` as an API at all.

Reply:
With all the current changes in the LSP context, maybe now is a good time to reconsider these two methods in `IN4JSCore` in general. I think their intended use case is still valid, but maybe there's a better way to do it. Let's discuss this on Monday, also with @mmews-n4 (no need to postpone this PR, though).

Reply:
I created ticket #1586